lib/Cake/Test/Fixture/BookFixture.php

```diff
@@ -43,6 +43,7 @@ class BookFixture extends CakeTestFixture {
  * @var array
  */
 	public $records = array(
-		array('id' => 1, 'isbn' => '1234567890', 'title' => 'Faust', 'author' => 'Johann Wolfgang von Goethe')
+		array('id' => 1, 'isbn' => '1234567890', 'title' => 'Faust', 'author' => 'Johann Wolfgang von Goethe'),
+		array('id' => 0, 'isbn' => '0000000001', 'title' => 'Bible', 'author' => 'Twelve Apostles'),
```

Comment:
It seems the problem is at the point of inserting the records. If we switch the positions of these two records we will get a duplicate primary key exception. I suspect that somewhere `!empty($record['id'])` is used instead of `!isset($record['id'])`. I'll try to dig deeper tonight or during the weekend.